packages/@jsii/kernel/lib/kernel.ts

```diff
@@ -148,6 +136,25 @@ export class Kernel {
     };
   }
 
+  public invokeBinScript(
+    req: api.InvokeScriptRequest,
+  ): api.InvokeScriptResponse {
+    const result = cp.spawnSync(
+      'node',
+      [path.join(this._getPackageDir(req.pkgname), req.script)].concat(
```

Comment:
I think the `script` value provided should not be a path, instead it should be a key from the `bin` map that was gathered from `package.json` during compilation.

You need to validate the script name was in the map, and fail if not (you don't want to allow a request to run *anything from anywhere*, that's kind of a bad security posture to have. Then, you need to use the *value* that corresponds to the `bin` key that was requested in order to compose the final executable path.

*Maybe* we shouldn't actually use `node` shell out here... Your NPM package's `bin` script might not actually be a node script (could be shell script, could be something else). On the other hand it might be tricky to make this work smooth on Windows if we skip the shell out (I guess it's a good first step?)